Use environment variable to override multiproc

fmaussion opened the pull request:
<!--
Thank you for your pull request!


Below are a few things we ask you kindly to self-check before (or during)
the process!
 
Remove checks that are not relevant.
-->

Closes https://github.com/OGGM/oggm/issues/936

- [x] Tests added/passed
- [ ] Fully documented
- [ ] Entry in `whats-new.rst`

TimoRoth commented on oggm/workflow.py:
```diff
@@ -147,7 +147,7 @@ def execute_entity_task(task, gdirs, **kwargs):
             return ogmpi.mpi_master_spin_tasks(pc, gdirs)
 
-    if cfg.PARAMS['use_multiprocessing']:
+    if cfg.PARAMS['use_multiprocessing'] and cfg.PARAMS['mp_processes'] > 1:
```
The issue here is that there are two other ways it could be 1.
Once if the system actually only has a single CPU, and one if the slurm env var is 1.

fmaussion replied:
> Once if the system actually only has a single CPU, and one if the slurm env var is 1.

But in this cases, we also would like to turn off multiprocessing altogether right? I thought this was the point of your comment on slack yesterday

TimoRoth replied:
Yes, that's what I'm saying.
If the CPU count is set to 1, there is no reason to even bother with multiprocessing.
And mp_processes is not the only way it can be set to 1.

fmaussion commented:
I have now moved the logic to cfg.initialize, so that mp_processes is decided at the very beginning. 

One question though: do you know how many CPUs will be given by `cpu_count` on Travis?

fmaussion commented:
I can't see from the travis log if multiproc was used or not, I think I need to add a check to the tests or something...

fmaussion commented:
OK I removed the "less than 1 process" magic, I think users should simply decide whether to use multiprocessing or not. 

I have however removed the spawn context per default, and added an env variable to switch it on.

@TimoRoth thoughts?

TimoRoth commented:
Looks good to me. I will see if I can somehow hot-patch this in the Anaconda-Path of things, because on Anaconda this is for some reason way more likely to cause problems.
